stingray/lightcurve.py

```diff
@@ -921,6 +921,55 @@ def _truncate_by_time(self, start, stop):
 
         return self._truncate_by_index(start, stop)
 
+    def split(self, min_gap, min_points=1):
+        """
+        For data with gaps, it can sometimes be useful to be able to split
+        the light curve into separate, evenly sampled objects along those
+        data gaps. This method allows to do this: it finds data gaps of a
+        specified minimum size, and produces a list of new `Lightcurve`
+        objects for each contiguous segment.
+
+        Parameters
+        ----------
+        min_gap : float
+            The length of a data gap, in the same units as the `time` attribute
+            of the `Lightcurve` object. Any smaller gaps will be ignored, any
+            larger gaps will be identified and used to split the light curve.
+
+        min_points : int, default 1
+            The minimum number of data points in each light curve. Light
+            curves with fewer data points will be ignored.
+
+        Returns
+        -------
+        lc_split : iterable of `Lightcurve` objects
+            The list of all contiguous light curves
+
+        Example
+        -------
+        >>> time = np.array([1, 2, 3, 6, 7, 8, 11, 12, 13])
+        >>> counts = np.random.rand(time.shape[0])
+        >>> lc = Lightcurve(time, counts)
+        >>> split_lc = lc.split(1.5)
+
+        """
+        # calculate the difference between time bins
+        tdiff = np.diff(self.time)
+        # find all distances between time bins that are larger than `min_gap`
+        gap_idx = np.where(tdiff >= min_gap)[0]
+
+        # tolerance for the newly created GTIs: Note that this seems to work with a 
+        # tolerance of 2, but not if I substitute 10, and I don't know why
+        epsilon = np.min(tdiff)/2.0
+
+        # calculate new GTIs
+        gti_start = np.hstack([self.time[0]-epsilon, self.time[gap_idx+1]-epsilon])
+        gti_stop = np.hstack([self.time[gap_idx]+epsilon, self.time[-1]+epsilon])
+        self.gti = np.vstack([gti_start, gti_stop]).T
```

Comment:
This seems to ignore existing GTIs in the lightcurve. If GTIs have not been applied yet, you might not see a gap in the time bins.
If this is the case, I would suggest to do something like
```
gti = np.vstack([gti_start, gti_stop]).T
if hasattr(self, 'gti') and self.gti is not None:
    gti = cross_two_gtis(self.gti, gti)
self.gti = gti
```